src/azure-cli/azure/cli/command_modules/cdn/commands.py

```diff
@@ -90,6 +91,30 @@ def _inner_not_found(ex):
                                  custom_func_name='update_endpoint',
                                  doc_string_source='azure.mgmt.cdn.models#EndpointUpdateParameters',
                                  supports_no_wait=True)
+        g.generic_update_command('add-rule', setter_name='update', setter_arg_name='endpoint_update_properties',
```

Comment:
[Command Guidline](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md#command-naming-and-behavior) 

According to your code, command should force to design like this:
```
az cdn rule add/list/remove
az cdn condition add/list/remove
az cdn action add/list/remove
```

Reply:
Hi, rule, conditions and actions are child collection of endpoint which has create/list/delete command. According to [this](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md#command-naming-and-behavior), for commands which maintain child collections, the follow pairings are acceptable.
CREATE/DELETE (same as top level resources)
ADD/REMOVE.
Do I still need to add the list command which will be duplicate of endpoint show

Reply:
Yeah. We would recommend this way even though it's might a little deep nested.

`CREATE/DELETE` usually used in create/delete a single resource.
`ADD/REMOVE` usually used in add/remove item in array.

If there is only one command for this subresource, we prefer `add-condition`, `add-rule`, `add-action`. Like `az network list-service-aliases`
But if there are more than one commands, we recommend do it like that doc suggested. 
> If a command subgroup would only have a single command, move it into the parent command group and hyphenate the name. This is common for commands which exist only to pull down cataloging information (e.g. database list-sku-definitions instead of database sku-definitions list). However, if you plan to add commands to this group in the future, then temporarily having a command group with a single command would make more sense. 

Such as:
```
az network application-gateway waf-policy -h

Group
    az network application-gateway waf-policy : Manage application gateway web application firewall
    (WAF) policies.

Subgroups:
    custom-rule    : Manage application gateway web application firewall (WAF) policy custom rules.
    managed-rule   : Manage managed rules of a waf-policy. Visit: https://docs.microsoft.com/en-
                     us/azure/web-application-firewall/afds/afds-overview.
    policy-setting : Defines contents of a web application firewall global configuration.

Commands:
    create         : Create an application gateway WAF policy.
    delete         : Delete an application gateway WAF policy.
    list           : List application gateway WAF policies.
    show           : Get the details of an application gateway WAF policy.
    update         : Update an application gateway WAF policy.
    wait           : Place the CLI in a waiting state until a condition of the application gateway
                     WAF policy is met.
```

Reply:
currently, it should be `rule add` instead of `add-rule`

Reply:
@MyronFanQiu  can you give me an example that I can follow?

Reply:
@haroldrandom I have the refactor the command. Now the command is :  
rule          add/list/remove
rule condition add/list/remove
rule action       add/list/remove